File: projects/engineering.purdue.edu/scriptingwithobjects/swocode/chap13/CascadingMenu.py
#!/usr/bin/env python

###  CascadingMenu.py

from Tkinter import *                 

mw = Tk()                                                            #(A)
text = Text( mw, width = 50 )                                        #(B)
text.pack()                                                          #(C)

mbar = Menu( mw )                                                    #(D)
mw.configure( menu = mbar )                                          #(E)

# Two menu buttons for the menubar:
level_0_menu = Menu( mbar, tearoff = 1 )                             #(F)
mbar.add_cascade( label = "Level_0",                                 #(G)
                  menu = level_0_menu,                               #(H)
                  underline = 1)                                     #(I)
helpmenu = Menu( mbar, tearoff = 0 )                                 #(J)
mbar.add_cascade( label = "Help",                                    #(K)
                  menu = helpmenu,                                   #(L)
                  underline = 0)                                     #(M)

# The menu for the 'Help' menubutton in the menubar:
helpmenu.add_command( label = "About",                               #(N)
                      command = lambda: text.insert('end',           #(O)
             "Click on the Level_0 button to see a cascading menu")) #(P)

# The menu for the 'Level_0' menubutton in the menubar:
level_0_menu.add_command( label = "Exit",                            #(Q)
                          command = mw.quit );                       #(R)

level_1_menu = Menu( mbar, tearoff = 1 )                             #(S)
level_0_menu.add_cascade( label = "Level_1",                         #(T)
                          underline = 1,                             #(U)
                          menu = level_1_menu )                      #(V)

level_2_menu =  Menu( mbar, tearoff = 1 )                            #(W)
level_1_menu.add_cascade( label = "Level_2",                         #(X)
                          underline = 2,                             #(Y)
                          menu = level_2_menu )                      #(Z)

level_3_menu =  Menu( mbar, tearoff = 1 )                            #(a)
level_2_menu.add_cascade( label = "Level_3",                         #(b)
                          underline = 3,                             #(c)
                          menu = level_3_menu )                      #(d)

level_4_menu =  Menu( mbar, tearoff = 1 )                            #(e)
level_3_menu.add_cascade( label = "Level_4",                         #(f)
                          underline = 4,                             #(g)
                          menu = level_4_menu )                      #(h)
level_4_menu.add_command( label = "click here",                      #(i)
                          command = lambda: text.insert( 'end',      #(j)
                 "hello from the tail end of a cascaded menu\n" ) )  #(k)
mainloop()                                                           #(l)

File: projects/demos/engineering.purdue.edu/scriptingwithobjects/swocode/chap13/CascadingMenu.pl
#!/usr/bin/perl -w

###  CascadingMenu.pl

use Tk;

my $mw = new MainWindow;                                             #(A)
my $txt = $mw->Scrolled( 'Text', -width => 50, -scrollbars => 'e' )  
             ->pack();                                               #(B)
my $mbar = $mw->Menu();                                              #(C)
$mw->configure( -menu => $mbar );                                    #(D)

# Two menubuttons for the menubar:

my $help = $mbar->cascade( -label => "Help",                         #(E)
                           -underline => 0,                          #(F)
                           -tearoff => 0 );                          #(G)

my $level_0 = $mbar->cascade( -label => "Level_0",                   #(H)
                              -underline => 0,                       #(I)
                              -tearoff => 1, );                      #(J)

# The menu for the 'Help' menubutton in the menubar:

$help->command( -label => "About",                                   #(K)
                 -command => sub { $txt->insert( 'end',              #(L)
      "Click on the Level_0 button to see a cascading menu" ) } );   #(M)

# The menu for the 'Level_0' menubutton in the menubar:

$level_0->command( -label => "Exit",                                 #(N)
                   -command => sub { exit } );                       #(O)

my $level_1 = $level_0->cascade( -label => "Level_1",                #(P)
                               -underline => 1,                      #(Q)
                               -tearoff => 1 );                      #(R)

my $level_2 = $level_1->cascade( -label => "Level_2",                #(S)
                               -underline => 2,                      #(T)
                               -tearoff => 1 );                      #(U)

my $level_3 = $level_2->cascade( -label => "Level_3",                #(V)
                               -underline => 3,                      #(W)
                               -tearoff => 1 );                      #(X)

my $level_4 = $level_3->cascade( -label => "Level_4",                #(Y)
                               -underline => 4,                      #(Z)
                               -tearoff => 1 );                      #(a)

$level_4->command( -label => "click here",                           #(b)
                  -command => sub { $txt->insert( 'end',             #(c)
                 "hello from the tail end of a cascaded menu\n" ) });#(d)
MainLoop;                                                            #(e)
